### PR TITLE
[Keyboard] Add Pachi RGB Rev 2

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Pachi RGB Rev 2 - duplex matrix and using L422KB instead of L422CB (rev1)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Include Pachi RGB Rev 2

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/xelus/pachi/rgb/rev1/config.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define ISSI_DRIVER_TOTAL DRIVER_LED_TOTAL
+
+#define RGB_MATRIX_STARTUP_VAL 80
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+
+#define FORCE_NKRO
```
Comment (drashna) [inline keyboards/xelus/pachi/rgb/rev1/config.h:75]: If you could: 
```suggestion

#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH

#define FORCE_NKRO
```

File: keyboards/xelus/pachi/rgb/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev2/rules.mk:21]:
```suggestion
```

File: keyboards/xelus/pachi/rgb/rev2/rev2.c [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+#include "rgb.h"
+#include <hal.h>
+#include <ch.h>
+
+// tested and working
+void matrix_io_delay(void) { __asm__ volatile("nop\nnop\nnop\n"); }
+
+#ifdef RGB_MATRIX_ENABLE
+#include <i2c_master.h>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev2/rev2.c:25]:
```suggestion
#include "i2c_master.h"
```

File: keyboards/xelus/pachi/rgb/rev1/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    make xelus/pachi/rgb/rev1:default
+
+Reset your keyboard in 3 ways:
+<ol>
+<li>Software reset on Fn + Backspace</li>
+<li>Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard</li>
+<li>Physical reset button: on the back of the PCB, there should be a small golden button you can press</li>
+</ol>
+
```
Comment (drashna) [inline keyboards/xelus/pachi/rgb/rev1/readme.md:18]:
```suggestion
* Software reset on Fn + Backspace
* Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard
* Physical reset button: on the back of the PCB, there should be a small golden button you can press

```

File: keyboards/xelus/pachi/rgb/rev2/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    make xelus/pachi/rgb/rev2:default
+
+Reset your keyboard in 3 ways:
+<ol>
+<li>Software reset on Fn + Backspace</li>
+<li>Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard</li>
+<li>Physical reset button: on the back of the PCB, there should be a small golden button you can press</li>
+</ol>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev2/readme.md:17]:
```suggestion

* Software reset on Fn + Backspace
* Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard
* Physical reset button: on the back of the PCB, there should be a small golden button you can press
```

File: keyboards/xelus/pachi/rgb/rev1/rev1.c [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+#include "rgb.h"
+#include <hal.h>
+#include <ch.h>
+
+// tested and working
+void matrix_io_delay(void) { __asm__ volatile("nop\nnop\nnop\n"); }
+
+#ifdef RGB_MATRIX_ENABLE
+#include <i2c_master.h>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev1/rev1.c:25]:
```suggestion
#include "i2c_master.h"
```

File: keyboards/xelus/pachi/rgb/rev1/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    make xelus/pachi/rgb/rev1:default
+
+Reset your keyboard in 3 ways:
+<ol>
+<li>Software reset on Fn + Backspace</li>
+<li>Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard</li>
+<li>Physical reset button: on the back of the PCB, there should be a small golden button you can press</li>
+</ol>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev1/readme.md:17]:
```suggestion

* Software reset on Fn + Backspace
* Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard
* Physical reset button: on the back of the PCB, there should be a small golden button you can press
```

File: keyboards/xelus/pachi/rgb/rev2/config.h [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define ISSI_DRIVER_TOTAL DRIVER_LED_TOTAL
+
+#define RGB_MATRIX_STARTUP_VAL 80
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+
+#define FORCE_NKRO
```
Comment (drashna) [inline keyboards/xelus/pachi/rgb/rev2/config.h:75]:
```suggestion
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH


#define FORCE_NKRO
```

File: keyboards/xelus/pachi/rgb/rev2/rev2.c [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rgb.h"
+#include <hal.h>
+#include <ch.h>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev2/rev2.c:19]:
```suggestion
#include "rev2.h"
```

File: keyboards/xelus/pachi/rgb/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L422
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (drashna) [inline keyboards/xelus/pachi/rgb/rev2/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/xelus/pachi/rgb/rev1/rev1.c [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rgb.h"
+#include <hal.h>
+#include <ch.h>
```
Comment (fauxpark) [inline keyboards/xelus/pachi/rgb/rev1/rev1.c:19]:
```suggestion
#include "rev1.h"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**